src/cloudWatchLogs/commands/viewLogStream.ts

```diff
@@ -110,21 +129,55 @@ export class DefaultSelectLogStreamWizardContext implements SelectLogStreamWizar
 
         const val = picker.verifySinglePickerOutput(choices)
 
-        let result = val?.label
+        const result = val?.label
+
+        // If the user wants to search the log group instead we return a different result
+        // compared to if they actually selected a log stream
+        return this.doSearchLogGroupProcess(result) ?? this.resolveSelectedLogStream(result)
+    }
+
+    /**
+     * Returns an object indicating the user wants to search the log groups,
+     * otherwise undefined
+     */
+    doSearchLogGroupProcess(quickPickResult?: string): LogSearchChoice | undefined {
```

Comment:
If we add indirection like this, the function name should be intuitive, else what's the purpose? This function name is inscrutable, so maybe the indirection should be dropped.